receiver/couchbasereceiver/README.md

```diff
@@ -0,0 +1,41 @@
+# CouchDB Receiver
+
+This receiver fetches stats from a couchbase cluster using the following endpoints:
+- `/pools/default` [endpoint](https://docs.couchbase.com/server/6.5/rest-api/rest-cluster-details.html)
+- `/pools/default/buckets/{bucket_name}/stats` [endpoint](https://docs.couchbase.com/server/6.5/rest-api/rest-bucket-stats.html).
+
+Supported pipeline types: `metrics`
+
+> :construction: This receiver is in **BETA**. Configuration fields and metric data model are subject to change.
+## Prerequisites
+
+This receiver supports Couchbase versions `6.5`, `6.6`, and `7.0`.
+
+## Configuration
+
+The following settings are required:
+- `endpoint` (default: `http://localhost:8091`): The URL of any node within the cluster to be monitored.
+- `username`
+- `password`
+
+The following settings are optional:
+
+- `collection_interval` (default = `10s`): This receiver collects metrics on an interval. Valid time units are `ns`, `us` (or `µs`), `ms`, `s`, `m`, `h`.
+- `tls` (defaults defined [here](https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/configtls/README.md)): TLS control. By default insecure settings are rejected and certificate verification is on.
+
+### Example Configuration
+
+```yaml
+receivers:
+  couchbase:
+    endpoint: http://localhost:8901
+    username: otelu
```

Comment:
It looks like couchbase also supports certificate auth

https://docs.couchbase.com/server/current/learn/security/authentication-overview.html#introduction-to-certificate-based-authentication

How about nesting like

```
password_auth:
  username: otelu
  password: $COUCHBASE_PASSWORD
```

to allow adding an alternative `certificate_auth` in the future?

Reply:
How bad would it be to not have auth defined here for this iteration? We could implement a basic auth authenticator extension, which would work for all HTTP clients built with `confighttp.HTTPClientSettings`. 

cc @pavankrish123

Reply:
The most common couchbase install will have a user/pass based authentication, so not including it in the initial version of the receiver would mean it won't work in most cases.

I do like the idea of a common auth implementation for HTTP clients. Would a basic auth authenticator extension require a specific yaml structure or would it be on the implementation's config parser to populate it? I'm wondering if we can code this in a way to make it not a breaking change when we do have a basic auth authenticator extension.

All that being said @anuraaga I'm also not apposed to creating a nested `password_auth` section. @jpkrohling Do you have any objections to that for this first iteration?

Reply:
I think it's important to support basic auth in the first iteration, because as @cpheps mentioned, that is the most typical configuration. In the long term we should definitely support the auth extension pattern, so I think this is largely a question of eventual backwards compatibility. To that point, I wonder if it's unreasonable to support both mechanisms indefinitely. Codewise, this seems easy enough. From a user standpoint, I think it provides an easier entry point for new users to try out the receiver. They can migrate to the auth extension later if they want. 

All that said, I haven't been involved in the auth extension design much, so I could be off base here. Curious @jpkrohling if you think it'd be acceptable to keep creds localized for now and also support an auth extension later.

Reply:
@jpkrohling I get what you're saying about using the authenticator and I agree with it. I think though as it doesn't currently exists and in the interest of time I would like to move forward with just having username/pass set directly. It seems the migration to this type of authenticator wouldn't be a heavy lift in the future.

@anuraaga To address the nesting, I did some thinking about it and looking at couchbases auth settings. Being that user/pass is the common case it would be more user friendly to have that at a top level. When/if we support certificates or domain auth the validation logic gets a bit more complex, regardless of how we make the config, but I think with good documentation and examples it we can make it simple for a user to use the different auth methods without nesting.

Reply:
@anuraaga, I slightly prefer the current implementation. I don't think we need the additional structure in order to allow alternate authentication mechanisms. We could quite easily add a `certificate` field later, along with config validation logic enforcing that the user specifies either `certificate` or `username/password`. In my opinion, this is a better user experience too, as it optimizes for the typical case while still providing immediate feedback if misconfigured.

Do you feel strongly about having separate auth structures?